File: Swift-Solutions/Problem-10.swift
// Find the sum of all the primes below two million.
import Foundation

func isPrime(num: Int) -> Bool {
  if num < 2 {
    return false
  }
  if num == 2 {
    return true
  }
  let limit: Int = Int(pow(Double(num),0.5))+1
  for i in 2...limit {
    if num % i == 0 {
      return false
    }
  }
  return true
}

func sumOfPrimes(limit: Int) -> Int {
  var sum: Int = 2
  var i: Int = 3
  while i < limit {
    if isPrime(num: i) {
      sum += i
    }
    i += 2
  }
  return sum
}

print(sumOfPrimes(limit: 2000000))

File: C++ Solutions/Problem-10.cpp
/* 
Find the sum of all the primes below two million. 
*/

#include <iostream>
using namespace std;

bool isPrime(double num);
int primeSum(int limit);

int main()
{
  int limit;
  cout << "Find the sum of all primes below ...\n";
  cin >> limit;
  int answer = primeSum(limit);
  cout << answer;
  return 0;
}

bool isPrime(double num) {
  if (num <= 1) {
    return false;
  }
  if (num == 2) {
    return true;
  }
  for (int i=2;i<int(sqrt(num))+1;i++) {
    if (int(num) % i == 0) {
      return false;
    }
  }
  return true;
}

int primeSum(int limit) {
  int sum = 2;
  for (int i=3; i<limit; i+=2) {
    if (isPrime(i)) {
      sum += i;
    }
  }
  return sum;
}
